src/dotty/tools/dotc/core/Contexts.scala

```diff
@@ -615,11 +615,22 @@ object Contexts {
 
   object Context {
 
+    // DO NOT MERGE INTO MASTER; this version is only intended
+    // to demonstrate that partest re-uses the same context base
+    // over multiple threads, which causes all sorts of races.
+    // Each thread in partest needs to start with its own context base.
+    @dotty.tools.sharable
+    private var thread: Thread = null
+
     /** Implicit conversion that injects all printer operations into a context */
     implicit def toPrinter(ctx: Context): Printer = ctx.printer
 
     /** implicit conversion that injects all ContextBase members into a context */
-    implicit def toBase(ctx: Context): ContextBase = ctx.base
+    implicit def toBase(ctx: Context): ContextBase = {
+      if (thread == null) thread = Thread.currentThread()
+      else assert(thread == Thread.currentThread(), "illegal multithreaded access to ContextBase")
```

Comment:
I'm not shure what is this test intended to demonstrate.
Currently it fails if more than 1 thread ever calls this method even if they use different `ContextBases`